### ISSUE TITLE
Can't change available current

### ISSUE THREAD

**Issue by pather79** [custom-components/zaptec#150]
Hi,

I just installed this addon in HA, and please let me know if I'm doing something wrong. :)

Anyway, when I try to set the available current, it does not work. When I have tried other things, like authorizing and it works. Below is the log (I changed something that might be a key in the API address to xxxxx):

Thank you!

Logger: homeassistant.components.websocket_api.http.connection
Källa: components/websocket_api/commands.py:245
integration: Home Assistant WebSocket API (dokumentation, ärenden)
Inträffade först: 09:01:11 (17 händelser)
Senast loggade: 14:49:19

[140311109270784] Unexpected exception
[140310815551648] Unexpected exception
Traceback (most recent call last):
  File "/config/custom_components/zaptec/number.py", line 60, in async_set_native_value
    await self.zaptec_obj.set_limit_current(availableCurrent=value)
  File "/config/custom_components/zaptec/api.py", line 445, in set_limit_current
    data = await self._account._request(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/zaptec/api.py", line 1005, in _request
    async for response, log_exc in ctx:
  File "/config/custom_components/zaptec/api.py", line 921, in _retry_request
    raise RequestRetryError(
custom_components.zaptec.api.RequestRetryError: Request to https://api.zaptec.com/api/installation/xxxxxxxxxxxxxxx/update failed after 5 retries

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 245, in handle_call_service
    response = await hass.services.async_call(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2802, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2845, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1007, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1079, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/number/__init__.py", line 122, in async_set_value
    await entity.async_set_native_value(native_value)
  File "/config/custom_components/zaptec/number.py", line 62, in async_set_native_value
    raise HomeAssistantError(f"Set current limit to {value} failed") from exc
homeassistant.exceptions.HomeAssistantError: Set current limit to 8.0 failed

**Comment by sveinse** [comment]
@pather79 what version of zaptec are you running?

The error is due to the integration not being able to call Zaptec cloud to set the value. Does it work setting the zaptec charge current in the Zaptec portal?

**Comment by pather79** [comment]
Thank you. Firmware in charger is 2.4.1.8. I can change it here:
![image](https://github.com/user-attachments/assets/829368c8-fdb9-4519-bf6f-b7381f79ac5f)

**Comment by pather79** [comment]
Oh, sorry. Zaptec integration in HA is 0.7.2. 
Yes, I can change to 8, it says successfully changed, and the 8 stays.

**Comment by sveinse** [comment]
I've seen the same thing in my logs:

```
Logger: custom_components.zaptec
Source: helpers/update_coordinator.py:414
integration: Zaptec EV charger (documentation)
First occurred: November 26, 2024 at 14:08:53 (2 occurrences)
Last logged: November 29, 2024 at 08:00:45

Error fetching xxx data: Request to https://api.zaptec.com/api/installation/xxx timed out after 5 retries
Error fetching xxx data: GET request to https://api.zaptec.com/api/installation/xxx failed with status 502: <ClientResponse(https://api.zaptec.com/api/installation/xxx) [502 Bad Gateway]> <CIMultiDictProxy('Server': 'nginx/1.25.3', 'Date': 'Fri, 29 Nov 2024 07:00:45 GMT', 'Content-Type': 'text/html', 'Content-Length': '157', 'Connection': 'keep-alive')>
```

Perhaps using only 5 retries is simply too little for Zaptec clould when it's busy? However, each time a request times out, it will wait a longer time (expoentially), so the updates will become very sluggish after retry number 5.

If the true cause here is the Zaptec cloud, there is very little I can do. Do you still experience the issue right now @pather79  ? I checked, and it works for me.

**Comment by pather79** [comment]
Yes, I still have the same error. I have tried several times during the day with same result. Strange.

**Comment by sveinse** [comment]
@pather79 What type of charger do you have?

**Comment by pather79** [comment]
Zaptec Go. I think I had problems with the charging button as well yesterday, but I did not try it enough to be sure that it’s a persistent problem.

**Comment by pather79** [comment]
I did some more testing and the Resume or Charging slider does not work either, if I try to initiate it from HA. If I manually start the charging in the Zaptec app, then the Pause, Resume and Charging buttons work. I have ECO mode enabled in the Zaptec app, meaning that it schedule the charging based on low cost hours. Don't know if that affects, but I have the "Start charging" button in the app even if Eco mode is enabled. And it should probably not affect the current setting from being changed. 

Authorization works perfect.

**Comment by sveinse** [comment]
I have no idea how Eco mode setting might affect the Zaptec integration. The only think the HA integration does is to send the commands to the Zaptec portal. There's no documentation that I have seen that describes if start or stop charging will work with Eco mode enabled.

**Comment by pather79** [comment]
It seems to be connected to the ECO mode. If I disable that, I can change the available current, and the start and resume buttons work. I guess we can close this issue, as it is working even if it would be nice to be able to override the ECO setting with a schedule in HA, but that is probably not related to this integration, I guess it's on Zaptec side.. A switch to enable/disable the ECO mode would also be nice to have in the integration, is that possible to add?

Thank you for your great work with this integration! :)

**Comment by sveinse** [comment]
I'm sure its possible to add the Eco mode switch. Although it's not documented by Zaptec, it is possible to reverse engineer it. However, the principal question becomes: How much control do we need or want in the HA integration? Especially for functions that Zaptec officially doesn't support? When is it better to rely on Zaptecs own portal?

**Comment by retifrav** [comment]
I got a similar/same(?) issue - trying to set the available charging current (*`zaptec_installation_available_current`*) I get these errors in the Home Assistant Core logs (*the ones viewable via Settings → System → Logs*):

```
Logger: homeassistant.components.websocket_api.http.connection
Source: components/websocket_api/commands.py:323
integration: Home Assistant WebSocket API (documentation, issues)
First occurred: 19:49:44 (7 occurrences)
Last logged: 20:33:21

[3590096408] Error during service call to number.set_value: Set current limit to 7.0 failed
[3673131160] Error during service call to number.set_value: Set current limit to 9.0 failed
[3549109096] Error during service call to number.set_value: Set current limit to 8.0 failed
[3879799096] Error during service call to number.set_value: Set current limit to 8.0 failed
[3556399656] Error during service call to number.set_value: Set current limit to 6.0 failed
```

And as @pather79 said, in my case it also turned out to be because I had Smart Eco Mode enabled in Zaptec Portal. As a matter of fact, I get the same problem when Charging mode in Zaptec Portal is set to any option other than Default Charging. And once I disabled Smart Eco Mode and set Charging mode to Default Charging, then I could change the available current from Home Assistant.

To summarize the requirements in a screenshot:

<img width="1444" height="602" alt="Image" src="https://github.com/user-attachments/assets/99596ecf-f337-41e6-a151-4f9099d80adb" />

It is a bit unfortunate that Smart Eco Mode has to be disabled, but I'd guess this is a limitation of Zaptec's API/backend, so there is nothing we can do about it.
